### DISCUSSION TITLE
Is there an example of a looping infinite scroll that fetches more from an API?

### DISCUSSION THREAD

**Discussion by molunika** [davidjerleke/embla-carousel#534] [Help]
I have been playing around with the infinite scroll examples and changing a few things, mainly turning on looping, changing the container to 3 columns (3 slides per view) and making the index start at 2. So if an array is [0,1,2,3,4,5], the viewport will show slides [1,2,3]. 

An idea would be to check if 0 or 5 is in view and then fetch more data from an API. I can really get it to reload the slides smoothly. 

Without looping, it's sooo much smoother.

**Comment by davidjerleke** [comment]
Hi @molunika,

Thank you for your question. This is an opinionated answer but I didn’t spend time to make the **Infinite scroll** example compatible with `loop: true`. This is because loading more slides while looping will disorient the user so I think it’s incredibly confusing UX. I would advise against that design choice for the very same reason.

With that said I think it’s possible to achieve smooth loading like the non loop setup with the Embla API. But this is unfortunately not prioritized among other things that are on the to do list. I hope you understand.

Best,
David
